File: algorithms/553. Optimal Division/solution.cc
class Solution {
public:
    string optimalDivision(vector<int>& nums) {
        if ( nums.size() == 1 ) {
            return to_string(nums[0]);
        }
        string result = to_string(nums[1]);
        for ( int i = 2; i < nums.size(); ++i ) {
            result += "/" + to_string(nums[i]);
        }
        if ( nums.size() > 2 ) {
            return to_string(nums[0]) + "/(" + result + ")";
        } else {
            return to_string(nums[0]) + "/" + result;
        }
    }
};

File: algorithms/553. Optimal Division/solution.py
class Solution(object):
    def optimalDivision(self, nums):
        """
        :type nums: List[int]
        :rtype: str
        """
        snums = map(str, nums)
        if len(nums) > 2:
            snums[1] = '(' + snums[1]
            snums[-1] = snums[-1] + ')'
        return '/'.join(snums)
